xonsh/environ.py

```diff
@@ -70,8 +79,28 @@ def current_branch(cwd=None, pad=True):
     return branch or ''
 
 
+@ensure_git
+def git_dirty(cwd=None):
+    """Returns a boolean as to whether there are uncommitted files on the
+    current branch of a git repository (if there is one)"""
+    try:
+        cmd = ['git', 'status', '--porcelain']
+        s = subprocess.check_output(cmd,
+                                    stderr=subprocess.PIPE,
+                                    cwd=cwd,
+                                    universal_newlines=True)
+        return bool(s)
+    except subprocess.CalledProcessError:
+        return False
+
+
+def git_dirty_color():
```

Comment:
+1 on the docstring.  Maybe `git_status_color` since the color is changed based on the status?